### PR TITLE
Development abstraction class cli demo

### PR DESCRIPTION
**Pull request by pp-pal**
- Add abstraction class for cli demo

### CODE REVIEW

File: typescript/node-js/cli/src/index.ts [hunk 1]
```diff
@@ -17,8 +17,9 @@
  * ---
  * Copyright (c) 2023 PT Darta Media Indonesia. All rights reserved.
  */
-import { Holder } from "./Holder";
-import { Issuer } from "./Issuer";
+import { MyHolder } from "./MyHolder";
+import { MyIssuer } from "./MyIssuer";
+import { MyVerifier } from "./MyVerifier";
 import {
   ISSUER_JKU,
```
Comment (LawrencePatrickSianto) [inline typescript/node-js/cli/src/index.ts:24]: remove unused imports

File: typescript/node-js/cli/src/libs/Holder.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ */
+import { ZkPassClient } from "@didpass/zkpass-client-ts";
+
+export abstract class Holder {
+  protected async createZkPassProof(
+    zkPassServiceUrl: string,
+    userDataToken: string,
+    dvrToken: string
+  ): Promise<string> {
+    console.log("\n#### starting zkpass proof generation...");
```
Comment (zulamdat) [inline typescript/node-js/cli/src/libs/Holder.ts:29]: @pp-pal ini bisa diganti sih lognya, soalnya sekarang double sama yg ada di MyHolder

**Review by zulamdat — APPROVED** [review]

File: typescript/node-js/cli/src/libs/Holder.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    // Step 2: Call the zkpassClient.generateZkpassProof
+    //         to get the zkpassProofToken.
+    //
+    const zkpassProofToken = await zkpassClient.generateZkpassProof(
+      userDataToken,
+      dvrToken
+    );
+
+    return zkpassProofToken;
+  }
```
Comment (ntejapermana) [inline typescript/node-js/cli/src/libs/Holder.ts:45]: Tambahin satu abstract function 
`public async start(
     userDataToken: string,
     dvrToken: string,
     zkPassServiceUrl: string,
     apiKey: ZkPassApiKey
   ): Promise<string>`

File: typescript/node-js/cli/src/libs/Issuer.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    //
+    const dataToken: string = await zkPassClient.signDataToJwsToken(
+      signingKey,
+      data,
+      verifyingKeyJws
+    );
+
+    return dataToken;
+  }
+}
```
Comment (ntejapermana) [inline typescript/node-js/cli/src/libs/Issuer.ts:54]: tambahin abstract function 
`public async getUserDataToken(dataFile: string): Promise<string>`

File: typescript/node-js/cli/src/libs/Verifier.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    // Step 4: Call zkPassClient.signToJwsToken.
+    // to digitally-sign the dvr data.
+    const dvrToken = await this.dvr.signToJwsToken(signingKey, verifyingKeyJws);
+    return dvrToken;
+  }
+
+  protected getDvr(): DataVerificationRequest | null {
+    return this.dvr;
+  }
+}
```
Comment (ntejapermana) [inline typescript/node-js/cli/src/libs/Verifier.ts:74]: Tambahin abstract function:
`async getDvrToken(dvrFile: string): Promise<string>`

**Review by ntejapermana — CHANGES_REQUESTED** [review]

File: typescript/node-js/cli/src/libs/Verifier.ts (hunk 5 above)
Comment (LawrencePatrickSianto) [reply typescript/node-js/cli/src/libs/Verifier.ts:76]: done, please recheck

File: typescript/node-js/cli/src/libs/Issuer.ts (hunk 4 above)
Comment (LawrencePatrickSianto) [reply typescript/node-js/cli/src/libs/Issuer.ts:56]: done, please recheck

File: typescript/node-js/cli/src/libs/Holder.ts (hunk 3 above)
Comment (LawrencePatrickSianto) [reply typescript/node-js/cli/src/libs/Holder.ts:45]: done, please recheck

**Review by ntejapermana — APPROVED** [review]

**Pull request merged**